app/scripts/lib/setupAccountLabelsPetnamesBridge.ts

```diff
@@ -0,0 +1,46 @@
+import { NameController, NameType } from '@metamask/name-controller';
+import {
+  PreferencesController,
+  AccountIdentityEntry,
+  PreferencesControllerState,
+} from '../controllers/preferences';
+import { NameEntry, OneWayNameBridge } from './NameBridge';
+
+export const ACCOUNT_LABEL_NAME_TYPE = NameType.ETHEREUM_ADDRESS;
+export const ACCOUNT_LABEL_VARIATION = '*';
+
+/**
+ * Sets up a bridge between the account labels in the preferences controller and
+ * the petnames in the name controller.
+ *
+ * @param preferencesController - The preferences controller to listen to.
+ * @param nameController - The name controller to update.
+ */
+export default function setupAccountLabelsPetnamesBridge(
+  preferencesController: PreferencesController,
+  nameController: NameController,
+) {
+  const bridge = new OneWayNameBridge(nameController, () =>
+    selectAccountLabelEntries(preferencesController.store.getState()),
+  );
+  preferencesController.store.subscribe(bridge.synchronize);
```

Comment:
In the abstract class approach, this preferences controller subscription would be encapsulated inside the `abstract onSourceChange` implementation for this child class.